api/common/interfaces.go

```diff
@@ -1154,6 +1157,34 @@ func getTopicBucketsCount(s Serverer, params map[string]interface{}) map[string]
 	return respPacking(count, SUCCESS)
 }
 
+// getID4ExtIP get ChordID correspond with external IP
+// params: ["address":<address>]
+// return: {"result":<result>, "error":<errcode>}
+func getID4ExtIP(s Serverer, params map[string]interface{}) map[string]interface{} {
+	if len(params) < 1 {
+		return respPacking(nil, INVALID_PARAMS)
+	}
+
+	addr, ok := params["RemoteAddr"].(string)
+	if !ok || len(addr) == 0 {
+		log.Errorf("Invalid params: [%v, %v]", ok, addr)
+		return respPacking(nil, INVALID_PARAMS)
+	}
+
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		if strings.LastIndexByte(addr, ':') < 0 { // Just only host, without port
+			host = addr
+		} else {
+			log.Errorf("getID4ExtIP met invalid params %v: %v", addr, err)
+			return respPacking(nil, INVALID_PARAMS)
+		}
+	}
+	id := address.GenChordID(fmt.Sprintf("%s:%d", host, config.Parameters.NodePort))
```

Comment:
I think we should use parsed port if addr contains one, if not we should use the default one like here. Otherwise it's not quite convenient in some case (local cluster for example)